### PR TITLE
Fix user agent in .net sdks

### PR DESCRIPTION
**Pull request by prathmesh-stripe**
Currently .NET SDKs rely on AssemblyName.Version to extract the version of Stripe SDK. This works for major releases but breaks for beta releases because how AssemblyVersion are calculated.
In a nutshell, a version `42.1.0-beta.1` would get converted to `42.1.0.0` in AssemblyName.Version. 

This new approach of storing SDK version as a constant is in-line with how we extract and update version in all our other SDKs.

### CODE REVIEW

File: src/StripeTests/StripeMockFixture.cs
Comment (jar-stripe) [inline src/StripeTests/StripeMockFixture.cs:113]: Given the issue in the PR description, does this properly keep the -beta or does this convert it to .0?

Comment (prathmesh-stripe) [reply src/StripeTests/StripeMockFixture.cs:113]: System.Version does not treat -beta properly. We don't need it since stripe-mock version does not have beta flag. 
I had to change this file was because of the addition of new Stripe.Version class. System.Version was being used earlier and this change just resolves ambiguity for the  compilers.

Comment (jar-stripe) [reply src/StripeTests/StripeMockFixture.cs:113]: Gotcha.  I would suggest leaving a comment in the code about this, but otherwise that makes sense.

**Review by jar-stripe — APPROVED** [review]
These changes look good to me.  Non blocking notes: I would add that comment explaining the limitation of System.Version and why it doesn't matter, and I _think_ that regex in the update-version make step is right but it would be worth testing it a few times before hitting the merge button.

**Pull request merged**
